h2o-py/h2o/frame.py

```diff
@@ -509,7 +509,10 @@ def logical_negation(self):
 
 
     def _unop(self, op):
-        return H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        ret = H2OFrame._expr(expr=ExprNode(op, self), cache=self._ex._cache)
+        if ret._ex._cache._names is not None:
+            ret._ex._cache._names = ["%s(%s)" % (op, name) for name in ret._ex._cache._names]
```

Comment:
Actually, should the types stay the same? For example, what if you conduct a `sin` on an int. Then wouldn't the `type` change?